stubs/tensorflow/tensorflow/data/__init__.pyi

```diff
@@ -223,8 +223,17 @@ class Dataset(ABC, Generic[_T1]):
         name: str | None = None,
     ) -> Dataset[Dataset[_T1]]: ...
     def with_options(self, options: Options, name: str | None = None) -> Dataset[_T1]: ...
+    @overload
+    @staticmethod
+    def zip(*args: Collection[Dataset[Any]], name: str | None = None) -> Dataset[tuple[Any, ...]]: ...
```

Comment:
The *args types here seem weird. In particular *args: tuple is rarely right choice. Even though *args is a tuple the type for it actually is type for values of tuple. A function that takes *args of one or more ints is *args: int not *args: tuple[int, …]

You could maybe use PEP 646 Unpack operator to specify a specific tuple like here.

Something like *args: Unpack[tuple[Dataset[T1], Dataset[T2]]]

Reply:
Yes, I copy/pasted it from the datasets variant without thinking :disappointed: 
Done in 8325466c6.